src/System.Text.Json/tests/Serialization/Stream.WriteTests.cs

```diff
@@ -60,6 +59,37 @@ public static async Task RoundTripAsync()
             }
         }
 
+        [Fact]
+        public static async Task RoundTripLargeJsonViaJsonElementAsync()
+        {
+            // Generating tailored json
+            int i = 0;
+            StringBuilder json = new StringBuilder();
+            json.Append("{");
+            while (true)
+            {
+                if (json.Length >= 14757)
+                {
+                    break;
+                }
+                json.AppendFormat(@"""Key_{0}"":""{0}"",", i);
+                i++;
+            }
+            json.Remove(json.Length - 1, 1).Append("}");
+
+            JsonElement root = JsonSerializer.Parse<JsonElement>(json.ToString());
+            var ms = new MemoryStream();
+            await JsonSerializer.WriteAsync(ms, root, root.GetType());
+        }
+
+        [Fact]
+        public static async Task RoundTripLargeJsonViaPocoAsync()
```

Comment:
@ahsonkhan I added new test with "poco", classic array...the issue doesn't raise up with "normal object" but only if we a [`IsValueConvertable`](https://github.com/dotnet/corefx/blob/a7ad25c7749cbd8ede10119a2e0ab43276192d8e/src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultConverters.cs#L20) object, this trigger https://github.com/dotnet/corefx/blob/4b6562e0da17ba1d92d72bc2406af9eea3e9337a/src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.cs#L34
that lead to exception below.